mattermost/mattermost-server
# [MM-42357] Add parameter to post search for current user mentions

#### Summary

Adds a boolean parameter `search_mentions` to the posts search endpoints:

```
/api/v4/posts/search
/api/v4/teams/<team_id>/posts/search
```

Only the first endpoint is intended to be used by the frontend to find recent mentions. 

Now, instead of providing the search terms for username and mention keys from the frontend, we can provide an empty search term and set `has_user_mention` to true. The username, mention keys, and user groups are automatically added to the search term.

QA Test Steps: 
- To test just the server change, make an API request with `search_mentions` set to `true` and `terms` set to the empty string. 
- Verify that the result has all the posts that mention the user according to the user's mention settings. 

The webapp PR is here: https://github.com/mattermost/mattermost-webapp/pull/11602. However the cloud servers are not building the branches together, so you will have to test locally if you want to test them together. 

#### Ticket Link

[Jira ticket](https://mattermost.atlassian.net/browse/MM-42357)

#### Release Note

```release-note
Added new parameter has_user_mention to API endpoint GET api/v4/posts/search and GET api/v4/teams/<team_id>/posts/search
```
